src/Microsoft.Extensions.DependencyModel/CompilationLibrary.cs

```diff
@@ -51,8 +51,32 @@ public class CompilationLibrary : Library
 
         public IEnumerable<string> ResolveReferencePaths()
         {
+            return ResolveReferencePaths(DefaultResolver);
+        }
+
+        public IEnumerable<string> ResolveReferencePaths(params ICompilationAssemblyResolver[] additionalResolvers)
```

Comment:
> The additional resolvers are called first, then the default resolver.
My bad, read the code wrong. 
> There should be no difference in perf between what is happening today and this code. Why the concern now?
New array and `CompositeCompilationAssemblyResolver` per invocation.